### ISSUE TITLE
`tilt up` without rebuilding container image

### ISSUE THREAD

**Issue by rodrigoscna** [tilt-dev/tilt#5821]
## Describe the Feature You Want
There could be a way to `tilt up` (with a flag or maybe an environment variable) that runs everything as it would, but without ever rebuilding the container images.
This would be useful for applications like Node.js that can reload based on file changes without the need to rebuild the whole container image.

### Current Behavior
As it is now, `tilt up` will always build the docker images.

### Why Do You Want This?
It would be useful specially for large monorepo applications where you can have your applications handle file changes without the need for a new container image to be rebuilt.

I am aware of the `file_sync_only` extension, but it requires some complex logic to be used with something like `helm_resource` (like rendering templates of the manifests to use in `file_sync_only`).

The proposal is something that does not requires any changes on the existing Tiltfiles, but rather just connect to existing deployed resources.

I'm not sure if it is even possible, since Tilt generates the image tag at the moment of the command, but maybe persisting the last used tag, or even connecting based solely on the image repository, without a tag.

**Comment by lizzthabet** [comment]
thanks for the feature request @rodrigoscna! tilt computes a content-based tag for each image, so as long as that doesn't change, tilt shouldn't be doing full rebuilds (even on `tilt up`). generally we see folks use image layer caching to minimize build time, but that may not have the impact you need in a large monorepo!

have you seen the [tiltfile config docs](https://docs.tilt.dev/tiltfile_config.html#specify-services-to-edit)? there's an example of defining a custom argument to `tilt` (in this case, it's called `to-edit`) and using the input to determine which `docker_build` calls to make. i wonder if that might get closer to the behavior you're looking for.

if you'd like to use the `file_sync_only` extension, you may also be able to use the [`helm` built-in](https://docs.tilt.dev/helm.html#iterate-on-chart-yaml) to grab a chart's yaml.

**Comment by rodrigoscna** [comment]
Thanks for the response @lizzthabet 
I'm actually already using image layer caching in my deployments, but it's not enough.

The thing is, I have tens of microservices being redeployed when I run `tilt up` and it takes a while for all of them to be ready.

But if Tilt was able to only connect to them and sync files from this moment on, it would basically be instantaneous.

I'm aware of the file_sync_only and the helm built-in. But they would demand some logic to be implemented on the Tiltfiles.
I'll give those a try, but it would be great if Tilt could have this feature built-in.

**Comment by nicks** [comment]
this discussion might be relevant: https://github.com/tilt-dev/tilt/issues/3282

i might quibble a bit with the statement "But if Tilt was able to only connect to them and sync files from this moment on, it would basically be instantaneous." An ad hoc filesync initialization step - making sure to both add new files and delete old ones - might not be instantaneous, and might even be slower than building a fresh immutable image. there's more back-and-forth communication, more edge cases, and fewer opportunities for caching.

But I appreciate your complaint that if you have tens of microservices, `tilt up` is still too slow.

**Comment by rodrigoscna** [comment]
I don't think it would be necessary to re-sync everything at the moment you start Tilt, only new changes/removals made after you start Tilt would be synced.

In any way, I was just making a feature suggestion.

**Comment by rodrigoscna** [comment]
Hello @grzegorznowak, I believe that strategy will help us for now, thank's for the suggestion.
